File: atcoder/177/C_Sum_of_product_of_pairs.py
def main():
    input()
    mod = 10**9 + 7
    arr = sorted(map(int, input().split()))
    arr.reverse()

    sm = arr[0] + arr[1]
    ans = arr[0] * arr[1] % mod

    for x in range(2, len(arr)):
        ans += sm * arr[x]
        sm += arr[x]

    print(ans % mod)


if __name__ == '__main__':
    main()

File: atcoder/ABC/177/C_Sum_of_product_of_pairs.js
const main = (input) => {
  input = input.trim().split('\n')
  const mod = BigInt(10**9 + 7)
  const arr = input[1].split(' ').map(BigInt)

  let sum = arr[0] + arr[1]
  let ans = arr[0] * arr[1] % mod

  for (let i=2; i<arr.length; i++) {
    ans += sum * arr[i] % mod
    sum += arr[i]
  }

  return console.log(Number(ans % mod))
}


process.env.MYTEST
  ? (process.env.MYTEST === 'test' ? test() : main(require('fs').readFileSync('dev/stdin', 'utf8')))
  : main(require('fs').readFileSync('/dev/stdin', 'utf8'))
